test/metabase/query_processor/pivot_test.clj

```diff
@@ -15,54 +16,167 @@
+                  "Source"
+                  "Created At"
+                  "pivot-grouping"
+                  "Count"]
+                 (map :display_name (mt/cols results))))
+        (is (apply distinct? rows))
+        ;; OLD RESULTS are commented out below
+        (is (= [;; ["Doohickey" "Facebook" "2019-01-01T00:00:00Z" 0  263 ]
+                ;; ["Doohickey" "Facebook" "2020-01-01T00:00:00Z" 0  89  ]
+                ;; ["Doohickey" "Google"   "2019-01-01T00:00:00Z" 0  276 ]
+                ;; ["Doohickey" "Google"   "2020-01-01T00:00:00Z" 0  100 ]
+                ;; ["Gizmo"     "Facebook" "2019-01-01T00:00:00Z" 0  361 ]
+                ;; ["Gizmo"     "Facebook" "2020-01-01T00:00:00Z" 0  113 ]
+                ;; ["Gizmo"     "Google"   "2019-01-01T00:00:00Z" 0  325 ]
+                ;; ["Gizmo"     "Google"   "2020-01-01T00:00:00Z" 0  101 ]
+                ;; [nil         nil        nil                    7  1628]
+                ["Doohickey" "Facebook" "2019-01-01T00:00:00Z" 0  263 ]
+                ["Doohickey" "Facebook" "2020-01-01T00:00:00Z" 0  89  ]
+                ["Doohickey" "Google"   "2019-01-01T00:00:00Z" 0  276 ]
+                ["Doohickey" "Google"   "2020-01-01T00:00:00Z" 0  100 ]
+                ["Gizmo"     "Facebook" "2019-01-01T00:00:00Z" 0  361 ]
+                ["Gizmo"     "Facebook" "2020-01-01T00:00:00Z" 0  113 ]
+                ["Gizmo"     "Google"   "2019-01-01T00:00:00Z" 0  325 ]
+                ["Gizmo"     "Google"   "2020-01-01T00:00:00Z" 0  101 ]
+                ;; [nil         "Facebook" "2019-01-01T00:00:00Z" 1  624 ]
+                ;; [nil         "Facebook" "2020-01-01T00:00:00Z" 1  202 ]
+                ;; [nil         "Google"   "2019-01-01T00:00:00Z" 1  601 ]
+                ;; [nil         "Google"   "2020-01-01T00:00:00Z" 1  201 ]
+                ;; ["Doohickey" nil        "2019-01-01T00:00:00Z" 2  539 ]
+                ;; ["Doohickey" nil        "2020-01-01T00:00:00Z" 2  189 ]
+                ;; ["Gizmo"     nil        "2019-01-01T00:00:00Z" 2  686 ]
+                ;; ["Gizmo"     nil        "2020-01-01T00:00:00Z" 2  214 ]
+                ;; [nil         nil        "2019-01-01T00:00:00Z" 3  1225]
+                ;; [nil         nil        "2020-01-01T00:00:00Z" 3  403 ]
+                ["Doohickey" "Facebook" nil                    4  352 ]
+                ["Doohickey" "Google"   nil                    4  376 ]
+                ["Gizmo"     "Facebook" nil                    4  474 ]
+                ["Gizmo"     "Google"   nil                    4  426 ]
+                ;; [nil         "Facebook" nil                    5  826 ]
+                ;; [nil         "Google"   nil                    5  802 ]
+                ["Doohickey" nil        nil                    6  728 ]
+                ["Gizmo"     nil        nil                    6  900 ]
+                [nil         nil        nil                    7  1628]]
+               rows))))))
```

Comment:
here's the results that were getting returned (I copied @paulrosenzweig 's example in #14329 directly) with the duplicate/unneeded rows commented out once I fixed the behavior